HW06.Rmd

```diff
@@ -1,92 +1,128 @@
+                         keytype = "TXID", columns = c("SYMBOL", "TXID"))
+
+ens.mapped_GBM <- left_join(ens.GBM.transcripts, map, by = c("gene_id" = "TXID")) %>% 
+  dplyr::select(-1) %>% 
+  dplyr::select(gene_symbol = SYMBOL, everything())
+
+ens.mapped_GBM <- ens.mapped_GBM[!duplicated(ens.mapped_GBM$gene_symbol),] #remove duplicated gene symbols
+  #these are removed instead of averaged because they simply do not correlate particularly well. 
+ens.mapped_GBM <- ens.mapped_GBM[!is.na(ens.mapped_GBM$gene_symbol),] #remove NA values
+```
+
+Do the t-test and make a table of the t-test results!
+
+```{r t.test for difference between a bunch of genes}
+#run this code to unload the libraries from before, it might be helpful because the select() function from dplyr might be hidden otherwise
+detach(package:EnsDb.Hsapiens.v75, unload = T)
+detach(package:ensembldb, unload = T)
+
+#add in your own gene of interest!!! 
+#I added 5S_rRNA :)
+genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2", "5S_rRNA")
+
+GBM.genes.of.interest <- filter(ens.mapped_GBM, gene_symbol %in% genes_of_interest)
+
+#Now perform a t-test between the H3K mutated and the wt samples. There are many ways to do this actually, you can use a for loop or you could do the tidy alternative with broom(), but the for loop is probably the easiest
+
+#to work in the tidyverse, it will be easier to make tidy the dataframe first
+
+tidy.GBM.genes.of.interest <- GBM.genes.of.interest %>% 
+  gather(key = "key", value = "value", -gene_symbol)  %>%
+  separate(key, into = c ("group", "order"), convert = TRUE)
+
+for (gene in genes_of_interest) {
+  gene_df <- tidy.GBM.genes.of.interest  %>%
+    filter(gene_symbol == gene)
+
+  #print out the t-test results
+  print(paste("========", gene, "========"))
+  print(t.test(value ~ group, gene_df))
+}
+```
+
+Now create a graphing function to create boxplots to visualize the results. Plot expression on the y-axis. The graph should look like [this example](IRX1_example.png)
+
+```{r create a graphing function}
+#create a graphing function
+graph_gene <- function(x) {
+  gene_df <- tidy.GBM.genes.of.interest  %>%
+    filter(gene_symbol == gene)
+  
+  p1 <- ggplot(gene_df, aes(group, value, fill = group)) + 
+    geom_boxplot() +
+    scale_x_discrete(labels = c('H3K27_Mutated','WT')) +
+    labs(title = paste(gene, "Expression in GBM models\n by H3K27 Mutated or WT Status"), 
+         x ="H3K27", y = "Expression_(TPM)") +
+    theme_bw() +
+    theme(legend.position = "none") +
+    theme(plot.title = element_text(hjust = 0.5))
+
+  print(p1)
+}
+
+#then use a for loop combined with the graphing function to make a graph for all your genes of interest 
+for (gene in genes_of_interest) {
+  graph_gene(gene)
+}
+```
```

Comment:
This was a very pleasant part, especially if the data were prepared correctly in the earlier parts. The function does its job and the graphs look as in the example. Overall great job. I have definitely learned a few new things from your assignment.